nbconvert/exporters/tests/test_templateexporter.py

```diff
@@ -114,6 +117,43 @@ def test_relative_template_file(self):
             exporter = self._make_exporter(config=config)
             assert os.path.abspath(exporter.template.filename) == template
             assert os.path.dirname(template) in [ os.path.abspath(d) for d in exporter.template_path ]
+    
+    def test_in_memory_template(self):
+        # Loads in an in memory template using jinja2.DictLoader
+        # creates a class that uses this template with the template_file argument
+        # converts an empty notebook using this mechanism
+        my_loader = DictLoader({'my_template': "{%- extends 'rst.tpl' -%}"})
+        
+        class MyExporter(TemplateExporter):
+            template_file = 'my_template'
+        
+        exporter = MyExporter(extra_loaders=[my_loader])
+        nb = v4.new_notebook()
+        out, resources = exporter.from_notebook_node(nb)
+
+
+    @pytest.mark.xfail(strict=True, raises=TemplateNotFound)
+    def test_in_memory_template_failure_to_find(self):
+        # Loads in an in memory template using jinja2.DictLoader
+        # creates a class that uses this template with the template_file argument
+        # converts an empty notebook using this mechanism
```

Comment:
Going from this commit message:

> Now it's happening in the Choice Loader and not the ExtensionTolerantLoader

I think I see the confusion. `ChoiceLoader` is what wraps the series of loaders to try, so that the user can provide `extra_loaders` with higher priority than our default loader. ChoiceLoader will try loaders in sequence. If one of those loaders throws `TemplateNotFound`, ChoiceLoader [catches that](https://github.com/pallets/jinja/blob/2.8/jinja2/loaders.py#L397) and moves on to try the next loader. If all loaders fail like that, ChoiceLoader finally raises TNF itself.

So the TNF exception you get from using our `TemplateExporter` class will always come from ChoiceLoader, because it catches the exceptions from individual loaders.